### DISCUSSION TITLE
Shadowing for Static meshes

### DISCUSSION THREAD

**Discussion by shukenmg** [microsoft/MixedReality-GraphicsTools-Unreal#9] [Q&A] [answered]
Would it be possible to support shadow casting on other static meshes that use GT_DefaultLit?
![image](https://user-images.githubusercontent.com/18000205/125200733-cc959f80-e29e-11eb-9f63-2a058f317f17.png)

Thanks!

**Comment by Cameron-Micka** [comment, marked as answer]
Hi @shukenmg, this is a great question. Unfortunately when a material's "Shading Model" property is marked as "Unlit" Unreal's renderer doesn't calculate any static (or dynamic) shadow casting for that mesh. Which makes sense in context. To circumvent this there are a few options you can try:

1) You could mark the material as "Default Lit" and "Fully Rough" and delete the "GT_DefaultLit" method. This makes a material that is cheap to render, and can cast/receive shadows. But... does not support any specular or environment reflections.

2) In the above example if the book and table are static in respect to each other you could bake shadows into the albedo texture(s) in another tool. Similar to doing lightmapping outside of Unreal.

3) You can use subtractive [proximity lights](https://github.com/microsoft/MixedReality-GraphicsTools-Unreal/blob/main/Docs/ProximityLights.md) to perform simple dynamic drop shadows. You can find an example in the `\GraphicsToolsProject\Plugins\GraphicsToolsExamples\Content\ProximityLights\ProximityLights.umap` level. 

![Shadow](https://user-images.githubusercontent.com/13305729/125322277-72bec500-e2f2-11eb-9277-ed8e75a43d34.gif)

**Reply by shukenmg** [reply]
Thank you very much, I'll try these ways.
